components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/TransactionHandlerComponent.java

```diff
@@ -0,0 +1,189 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.crypto.Cipher;
+import javax.sql.DataSource;
+
+/**
+ * This class represents the internal transaction count handler component.
+ */
+public class TransactionHandlerComponent {
+
+    private static final Log LOG = LogFactory.getLog(TransactionHandlerComponent.class);
+    private static final int DEFAULT_SCHEDULED_PERIOD_INTERVAL = 60;
+    private static final String SCHEDULED_PERIOD = "transactionCountScheduledPeriod";
+
+    private static final String TRANSACTION_DATA_SOURCE = "transactionDataSource";
+    private static final String TRANSACTION_CONFIG_SECTION = "transaction";
+    private static final String TRANSACTION_CONFIG_NAME = "data_source";
+
+    private ScheduledExecutorService trnCountWriterTaskScheduler;
+
+    public void start(DataSourceService dataSourceService) throws TransactionException, DataSourceException {
+        String nodeId = generateRandomId();
+
+        // initialize cipher for encryption needs.
+        Cipher cipher = CryptoUtil.initializeCipher();
+
+        DataSource dataSource = getTransactionDataSource(dataSourceService);
+
+        // initialize Transaction store.
+        TransactionStore transactionStore = new TransactionStore(dataSource, nodeId, cipher);
+
+        // start and schedule transaction writer task.
+        scheduleTransactionWriterTask(transactionStore);
+    }
+
+    /**
+     * Get the time rate for running the transaction writer scheduled task.
+     *
+     * @return - scheduled period.
+     */
+    private int getTransactionCountScheduledPeriod() {
+        String scheduledPeriodStr = System.getProperty(SCHEDULED_PERIOD);
+        if (StringUtils.isEmpty(scheduledPeriodStr)) {
+            scheduledPeriodStr = System.getenv(SCHEDULED_PERIOD);
+            if (StringUtils.isEmpty(scheduledPeriodStr)) {
+                Object scheduledPeriodObject = ConfigParser.getParsedConfigs().get("" + "." + "");
+                if (null != scheduledPeriodObject) {
+                    scheduledPeriodStr = scheduledPeriodObject.toString();
+                }
+            }
+        }
+        try {
+            return Integer.parseInt(scheduledPeriodStr);
+        } catch (NumberFormatException e) {
+            return DEFAULT_SCHEDULED_PERIOD_INTERVAL;
+        }
+    }
+
+    /**
+     * Schedule the transaction writer task to write the current transaction count to the database at a fixed rate.
+     *
+     * @param transactionStore - transactionStore instance.
+     */
+    private void scheduleTransactionWriterTask(TransactionStore transactionStore) {
+        trnCountWriterTaskScheduler = Executors.newSingleThreadScheduledExecutor();
+        int scheduledPeriod = getTransactionCountScheduledPeriod();
+
+        trnCountWriterTaskScheduler.scheduleAtFixedRate(() -> {
+            try {
+                transactionStore.addTransaction();
```

Comment:
Could there be any other type of exceptions? We need to make sure, this task will not get killed due to any reason (exception throw). It should log error and continue running.

Reply:
Shall we add a debug log here?

Reply:
Then, shall we catch the "Exception" here? WDYT?

Reply:
+1

Reply:
catching a Throwable here to avoid any possible exceptions (other than Errors) that can cause this task to terminate.